src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionCertificateStoreProvider.java

```diff
@@ -74,11 +101,123 @@ public String getName() {
         if (isWindows) {
             plainCek = decryptColumnEncryptionKeyWindows(masterKeyPath, encryptionAlgorithm,
                     encryptedColumnEncryptionKey);
-        } else {
+        } else if (null != keyStorePwd && null != keyStorePath) {
+            plainCek = decryptColumnEncryptionKeyNonWindows(masterKeyPath, encryptionAlgorithm, encryptedColumnEncryptionKey);
+        }  else {
             throw new SQLServerException(SQLServerException.getErrString("R_notSupported"), null);
         }
         windowsCertificateStoreLogger.exiting(SQLServerColumnEncryptionCertificateStoreProvider.class.getName(),
                 "decryptColumnEncryptionKey", "Finished decrypting Column Encryption Key.");
         return plainCek;
     }
+
+    private byte[] decryptColumnEncryptionKeyNonWindows(String masterKeyPath,
+                                                        String encryptionAlgorithm,
+                                                        byte[] encryptedColumnEncryptionKey) throws SQLServerException {

```

Comment:
**Reporter**: CHECKSTYLE
**Rule**: com.puppycrawl.tools.checkstyle.checks.FinalParametersCheck
**Severity**: ERROR
**File**: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerColumnEncryptionCertificateStoreProvider.java L116

Parameter encryptedColumnEncryptionKey should be final.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<849825521>*